pyk/src/pyk/k2lean4/model.py

```diff
@@ -171,6 +171,147 @@ def __str__(self) -> str:
         return '\n'.join(lines)
 
 
+@final
+@dataclass(frozen=True)
+class Instance(Declaration):
+    signature: Signature
+    val: DeclVal
+    attr_kind: AttrKind | None
+    priority: int | None
+    ident: DeclId | None
+    modifiers: Modifiers | None
+
+    def __init__(
+        self,
+        signature: Signature,
+        val: DeclVal,
+        attr_kind: AttrKind | None = None,
+        priority: int | None = None,
+        ident: str | DeclId | None = None,
+        modifiers: Modifiers | None = None,
+    ):
+        if not signature.ty:
+            # TODO refine type to avoid this check
+            raise ValueError('Missing type from signature')
+        ident = DeclId(ident) if isinstance(ident, str) else ident
+        object.__setattr__(self, 'signature', signature)
+        object.__setattr__(self, 'val', val)
+        object.__setattr__(self, 'attr_kind', attr_kind)
+        object.__setattr__(self, 'priority', priority)
+        object.__setattr__(self, 'ident', ident)
+        object.__setattr__(self, 'modifiers', modifiers)
+
+    def __str__(self) -> str:
+        modifiers = f'{self.modifiers} ' if self.modifiers else ''
+        attr_kind = f'{self.attr_kind.value} ' if self.attr_kind else ''
+        priority = f' priority := {self.priority}' if self.priority is not None else ''
+        ident = f' {self.ident}' if self.ident else ''
+        signature = f' {self.signature}' if self.signature else ''
+
+        decl = f'{modifiers}{attr_kind}instance{priority}{ident}{signature}'
+
+        match self.val:
+            case SimpleVal():
+                return f'{decl} := {self.val}'
```

Comment:
What kind of instance does this correspond to?

Reply:
For example: https://github.com/leanprover/lean4/blob/9dbe5e6f9ce6bc686541218f62ff6b8063dfc4dc/src/Lean/Data/AssocList.lean#L25

Reply:
I see, thank you!